sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClient.java

```diff
@@ -153,15 +163,47 @@ private static BiFunction<HttpClientRequest, NettyOutbound, Publisher<Void>> bod
                     hdr.getValuesList().forEach(value -> reactorNettyRequest.addHeader(hdr.getName(), value));
                 }
             }
-            if (restRequest.getBody() != null) {
-                Flux<ByteBuf> nettyByteBufFlux = restRequest.getBody().map(Unpooled::wrappedBuffer);
-                return reactorNettyOutbound.send(nettyByteBufFlux);
+            BinaryData body = restRequest.getBodyAsBinaryData();
+            if (body != null) {
+                BinaryDataContent bodyContent = BinaryDataHelper.getContent(body);
+                if (bodyContent instanceof ByteArrayContent) {
+                    return reactorNettyOutbound.send(Mono.just(Unpooled.wrappedBuffer(bodyContent.toBytes())));
+                } else if (bodyContent instanceof StringContent
+                    || bodyContent instanceof SerializableContent) {
+                    return reactorNettyOutbound.send(
+                        Mono.fromSupplier(() -> Unpooled.wrappedBuffer(bodyContent.toBytes())));
+                } else if (bodyContent instanceof FileContent) {
+                    FileContent fileContent = (FileContent) bodyContent;
+                    // Beware of NettyOutbound.sendFile(Path) it involves extra file length lookup.
+                    // This is going to use zero-copy transfer if there's no ssl
+                    return reactorNettyOutbound.sendFile(fileContent.getFile(), 0, fileContent.getLength());
```

Comment:
Does this take the `chunkSize` defined in [FileContent](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core/src/main/java/com/azure/core/implementation/util/FileContent.java#L44) into consideration when doing the transfer?

Reply:
I was under impression that this expanded logic was for less operators, but it was indeed to honor the chunk size. I'll bring it in sooner.
https://github.com/Azure/azure-sdk-for-java/blob/af866a67664c5bca1b4cb33ebcfbc19966439dc5/sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClient.java#L186-L214